src/renderer/components/Footer.tsx

```diff
@@ -26,8 +31,40 @@ export const BasicFooter = styled.div`
 `;
 
 export const Footer: React.FC = () => {
+  const { showInfo, showError } = useToasts();
+  const { dolphinService } = useServices();
+  const { checkForAppUpdates } = useAppUpdate();
+  const { updateDolphin } = useDolphinActions(dolphinService);
+  const [checkingForUpdates, setCheckingForUpdates] = React.useState(false);
+  const checkForUpdatesHandler = React.useCallback(async () => {
+    setCheckingForUpdates(true);
+    try {
+      showInfo("Checking for updates...");
+      await checkForAppUpdates();
+      await updateDolphin();
+    } catch (err) {
+      window.electron.log.error(err);
+      showError("Failed to get updates");
+    }
+    setTimeout(() => setCheckingForUpdates(false), 10000);
```

Comment:
```suggestion
    } finally {
      setCheckingForUpdates(false);
    }
```
Not sure what the `setTimeout` here is for, should this be a `finally`?

Reply:
the reason i put that there is to avoid people spamming the button cause
```
await checkForAppUpdates();
await updateDolphin();
```
return almost immediately rather than after the checks are done.

Reply:
That's almost certainly not the correct solution to the problem. To only trigger the event once, upon button spam completion you can use debounce. To trigger the event at most once every X during button spam you can use throttle. I think it's still probably correct to set the `setCheckingForUpdates` back to false on `finally` though.